src/client/runtimeSitePreview/PreviewSite.ts

```diff
@@ -0,0 +1,106 @@
+        return websiteDetails?.websiteUrl || "";
+    }
+
+    static async launchBrowserAndDevToolsWithinVsCode(webSitePreviewURL: string): Promise<void> {
+
+        const edgeToolsExtensionId = 'ms-edgedevtools.vscode-edge-devtools';
+        const edgeToolsExtension = vscode.extensions.getExtension(edgeToolsExtensionId);
+
+        if (edgeToolsExtension) {
+            // Preserve the original state of the launch.json file and .vscode folder
+            const workspaceFolder = vscode.workspace.workspaceFolders?.[0];
+            const vscodeFolderPath = workspaceFolder ? path.join(workspaceFolder.uri.fsPath, '.vscode') : null;
+            const launchJsonPath = vscodeFolderPath ? path.join(vscodeFolderPath, 'launch.json') : null;
+            let originalLaunchJsonContent: string | null = null;
+            let vscodeFolderExisted = false;
+
+            if (vscodeFolderPath && fs.existsSync(vscodeFolderPath)) {
+                vscodeFolderExisted = true;
+                if (launchJsonPath && fs.existsSync(launchJsonPath)) {
+                    originalLaunchJsonContent = fs.readFileSync(launchJsonPath, 'utf8');
+                }
+            }
+
+            await updateLaunchJsonConfig(webSitePreviewURL);
+
+            try {
+                // Added a 2-second delay before executing the launchProject command to handle the case where the launch.json file is not saved yet
+                await new Promise(resolve => setTimeout(resolve, 2000));
+                await vscode.commands.executeCommand('vscode-edge-devtools-view.launchProject');
+
+            } finally {
+                // Revert the changes made to the launch.json file and remove the .vscode folder if it was created
+
+                // Added a 2-second delay to ensure that debug session is closed and then launch.json file is removed
+                await new Promise(resolve => setTimeout(resolve, 2000));
+                if (launchJsonPath) {
+                    if (originalLaunchJsonContent !== null) {
+                        fs.writeFileSync(launchJsonPath, originalLaunchJsonContent, 'utf8');
+                    } else if (fs.existsSync(launchJsonPath)) {
+                        fs.unlinkSync(launchJsonPath);
+                    }
+                }
+
+                if (vscodeFolderPath && !vscodeFolderExisted && fs.existsSync(vscodeFolderPath)) {
+                    const files = fs.readdirSync(vscodeFolderPath);
+                    if (files.length === 0) {
+                        fs.rmdirSync(vscodeFolderPath);
+                    }
+                }
+            }
+        } else {
+            const install = await vscode.window.showWarningMessage(
+                vscode.l10n.t(
+                `The extension Microsoft Edge Tools is required to run this command. Do you want to install it now?`,
+                'Install', 'Cancel'
+            ));
```

Comment:
Is this working or do we need to pass `Install` and `Cancel` as arguments to `showWarningMessage`?